src/main/java/org/fsg1/fmms/backend/endpoints/SemestersEndpoint.java

```diff
@@ -38,7 +38,7 @@ public CurriculumEndpoint(final CurriculumService service) {
     @Path("/semesters")
     @Produces(MediaType.APPLICATION_JSON)
     public Response getCurriculumSemesters(@PathParam("curriculum_id") final int curriculumId) throws Exception {
-        final CurriculumService service = (CurriculumService) getService();
+        final SemestersService service = (SemestersService) getService();
```

Comment:
Can't you avoid casting by using Generic type parameter at base class?

Reply:
Done